code/utilities/helpers/EnvHelper.py

```diff
@@ -40,13 +42,17 @@ def __init__(self, **kwargs) -> None:
         self.AZURE_OPENAI_STREAM = os.getenv('AZURE_OPENAI_STREAM', '')
         self.AZURE_OPENAI_EMBEDDING_MODEL = os.getenv('AZURE_OPENAI_EMBEDDING_MODEL', '')
          # Set env for OpenAI SDK
-        self.OPENAI_API_TYPE = "azure"
         self.OPENAI_API_BASE = f"https://{os.getenv('AZURE_OPENAI_RESOURCE')}.openai.azure.com/"
-        self.OPENAI_API_KEY = self.AZURE_OPENAI_KEY
+        if self.AZURE_AUTH_TYPE == "keys": 
+            self.OPENAI_API_KEY = self.AZURE_OPENAI_KEY
+            self.OPENAI_API_TYPE = "azure"
+        else:
+            self.OPENAI_API_KEY = DefaultAzureCredential(exclude_shared_token_cache_credential=True).get_token("https://cognitiveservices.azure.com/.default").token
```

Comment:
I'm not sure having this logic in a generic `EnvHelper` is best. I think it could be confusing as sometimes the `OPENAI_API_KEY` will be set to a token instead of a key. 

Could we instead move this logic to the constructor of the `LLMHelper` which uses the key? What do you think?